backend/infrahub/database/__init__.py

```diff
@@ -312,27 +320,36 @@ async def execute_query_with_metadata(
         params: dict[str, Any] | None = None,
         name: str = "undefined",
         context: dict[str, str] | None = None,
+        type: QueryType | None = None,  # pylint: disable=redefined-builtin
     ) -> tuple[list[Record], dict[str, Any]]:
         with trace.get_tracer(__name__).start_as_current_span("execute_db_query_with_metadata") as span:
             span.set_attribute("query", query)
             if name:
                 span.set_attribute("query_name", name)
 
-            runtime = Neo4jRuntime.UNDEFINED
+            runtime = self.default_neo4j_runtime
+
+            if self.db_type == DatabaseType.NEO4J:
+                query_config = self.queries_names_to_config.get(name, None)
+
+                if query_config and query_config.neo4j_runtime not in [Neo4jRuntime.DEFAULT, Neo4jRuntime.UNDEFINED]:
+                    runtime = query_config.neo4j_runtime
```

Comment:
```suggestion

                if query_config := self.queries_names_to_config.get(name, None):
                    runtime = query_config.neo4j_runtime
```

Reply:
query_config is referenced below as well,I think this is best to keep it as a variable